sdk/python/feast/feature_service.py

```diff
@@ -54,18 +52,20 @@ def __init__(
             ValueError: If one of the specified features is not a valid type.
         """
         self.name = name
-        self.features = []
-        for feature in features:
-            if (
-                isinstance(feature, FeatureTable)
-                or isinstance(feature, FeatureView)
-                or isinstance(feature, OnDemandFeatureView)
+        self.feature_view_projections = []
+
+        for feature_grouping in features:
+            if isinstance(feature_grouping, FeatureTable):
+                self.feature_view_projections.append(
+                    FeatureViewProjection.from_definition(feature_grouping)
+                )
+            elif isinstance(feature_grouping, FeatureView) or isinstance(
+                feature_grouping, OnDemandFeatureView
             ):
-                self.features.append(FeatureViewProjection.from_definition(feature))
-            elif isinstance(feature, FeatureViewProjection):
-                self.features.append(feature)
+                self.feature_view_projections.append(feature_grouping.projection)
             else:
-                raise ValueError(f"Unexpected type: {type(feature)}")
+                raise ValueError(f"Unexpected type: {type(feature_grouping)}")
```

Comment:
Nitpick: Would it make sense to expand this exception to be more descriptive? I can imagine people running `apply` and seeing this exception and not knowing where the bad type is being used. What about `The feature service {fs_name} has been provided with an invalid type {type(feature_grouping)} as part of the \"features\" argument`?